src/main/java/mcscheduler/logic/commands/RoleEditCommand.java

```diff
@@ -0,0 +1,139 @@
+package mcscheduler.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import mcscheduler.commons.core.Messages;
+import mcscheduler.commons.core.index.Index;
+import mcscheduler.commons.util.CollectionUtil;
+import mcscheduler.logic.commands.exceptions.CommandException;
+import mcscheduler.model.Model;
+import mcscheduler.model.assignment.Assignment;
+import mcscheduler.model.role.Role;
+import mcscheduler.model.shift.RoleRequirement;
+import mcscheduler.model.shift.Shift;
+import mcscheduler.model.worker.Worker;
+
+/**
+ * Edits the role identified in the McScheduler.
+ */
+public class RoleEditCommand extends Command {
+
+    public static final String COMMAND_WORD = "role-edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the role identified by the index number used in the displayed role list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "ROLE (must be alphanumeric and can contain spaces)\n"
+            + "Example: " + COMMAND_WORD + " 1 burger flipper";
+
+    public static final String MESSAGE_EDIT_ROLE_SUCCESS = "Edited role: %1$s";
+
+    private final Index targetIndex;
+    private final Role editedRole;
+
+    /**
+     * Creates a {@code RoleEditCommand}.
+     *
+     * @param targetIndex of the role in the filtered role list to edit.
+     * @param editedRole updated role to replace the original in the model.
+     */
+    public RoleEditCommand(Index targetIndex, Role editedRole) {
+        this.targetIndex = targetIndex;
+        this.editedRole = editedRole;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Role> roleList = model.getFilteredRoleList();
+
+        if (targetIndex.getZeroBased() >= roleList.size()) {
+            throw new CommandException(
+                    String.format(Messages.MESSAGE_INVALID_ROLE_DISPLAYED_INDEX, targetIndex.getOneBased()));
+        }
+
+        Role roleToEdit = roleList.get(targetIndex.getZeroBased());
+        editRoleInShifts(model, roleToEdit);
+        editRoleInWorkers(model, roleToEdit);
+        editRoleInAssignments(model, roleToEdit);
```

Comment:
Please prevent leaves from being edited by this function or it will have the ability to remove all workers from leave (which I don't think is intended). Although since it's index based that should never happen, but some assertions always help just in case someone decides to be funny and edit the save file.

Reply:
Good point